src/Module/Trasher/Trasher.php

```diff
@@ -57,6 +70,10 @@ public function __construct( TrasherSettingRepository $setting_repository, Conte
 	 */
 	public function trash_related_posts( $post_id ): int {
 
+		if ( ! $this->active_post_types->includes( get_post_type( (int) $post_id ) ) ) {
```

Comment:
We don't cast anything that is only for WordPress - because it doesn't care anway. So no need for the `(in)` cast.
Also, in rare cases, `get_post_type()` might return `false`.
Thus, better cast to string: `(string) get_post_type( $post_id )`.
This is also true for the other two files (because `$post->post_type` is public, and might be anything).